pkg/controller/service/service_controller.go

```diff
@@ -676,7 +686,19 @@ func (s *ServiceController) lockedUpdateLoadBalancerHosts(service *v1.Service, h
 	// This operation doesn't normally take very long (and happens pretty often), so we only record the final event
 	err := s.balancer.UpdateLoadBalancer(s.clusterName, service, hosts)
 	if err == nil {
-		s.eventRecorder.Event(service, v1.EventTypeNormal, "UpdatedLoadBalancer", "Updated load balancer with new hosts")
+		// If there are no available nodes for LoadBalancer service, we need to update service status.
+		if len(hosts) == 0 {
+			previousState := v1helper.LoadBalancerStatusDeepCopy(&service.Status.LoadBalancer)
+			emptyState := &v1.LoadBalancerStatus{}
+			err = s.updateServiceStatus(service, previousState, emptyState)
+			if err != nil {
+				glog.Errorf("Failed to update the status of LoadBalancer service %s/%s", service.Namespace, service.Name)
+			}
+
+			s.eventRecorder.Eventf(service, v1.EventTypeWarning, "UnAvailableService", "There are no available nodes for LoadBalancer service %s/%s", service.Namespace, service.Name)
```

Comment:
Should it be `UnAvailableLoadBalancer` instead?

Reply:
I'm a bit worried about this event will become spam when there is no nodes, as we will be emitting this for all loadbalancer services.

Reply:
Yes, If  there is no nodes, these load balancer service should get this event.